tpmutil/encoding.go

```diff
@@ -0,0 +1,266 @@
+// Copyright (c) 2018, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tpmutil
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"reflect"
+)
+
+// LengthPrefixSize is the size in bytes of length prefix for byte slices.
+//
+// In TPM 1.2 this is 4 bytes.
```

Comment:
In cases like this, I find it useful to simplify the job of the user of this package. Since this code knows that this value should either be 4 bytes or 2 bytes, consider adding a function to set it and making this variable internal to the package. This makes client code clearer and reduces the chances of an error. E.g.,

```go
func SetTPM12LengthPrefixSize() {
  LengthPrefixSize = 4
}

func SetTPM20LengthPrefixSize() {
  LengthPrefixSize = 2
}
```

Reply:
Good idea, done